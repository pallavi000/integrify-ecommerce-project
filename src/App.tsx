import { BrowserRouter as Router } from "react-router-dom";
import { useEffect } from "react";

// Redux
import { useSelector } from "react-redux";
import { AppState, useAppDispatch } from "./redux/store";

// Reducer
import { getCurrentUser } from "./redux/reducers/authReducer";
import AppRouter from "./routes/AppRouter";

// components
import ScrollToTop from "./components/ScrollToTop";

const App = () => {
  // App Dispatch
  const dispatch = useAppDispatch();

  // Auth States
  const { access_token, user } = useSelector((state: AppState) => ({
    access_token: state.auth.access_token,
    user: state.auth.user,
  }));

  // Get current user
  useEffect(() => {
    if (access_token && !user) {
      dispatch(getCurrentUser());
    }
  }, [access_token, user, dispatch]);

  return (
    <Router>
      <ScrollToTop />
      <AppRouter />
    </Router>
  );
};

export default App;
