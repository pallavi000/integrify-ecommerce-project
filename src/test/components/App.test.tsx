// import { render, screen, waitFor } from "@testing-library/react"
// import { Provider } from "react-redux"
// import store from "../shared/store"
// import App from "../../App"
// import appRender from "./appRender"
// import userServer from "../servers/userServer"

import { waitFor, screen } from "@testing-library/react";
import App from "../../App";
import appRender from "./appRender";

// beforeAll(() => userServer.listen())
// afterAll(() => userServer.close())

describe("Test App component", () => {
  test("Should layout match snapshot", async () => {
    /* const { asFragment } = render(
            <Provider store={store}>
                <App />
            </Provider>
        ) */
    //const {baseElement, findByText} = appRender(<App />)
    // expect(baseElement).toMatchSnapshot()
    // expect(screen.getAllByText("Create new user").length).toBe(1)
    // appRender(<App />);
    // await waitFor(() => screen.getAllByText(/logo/i));
    // expect(screen.getAllByText(/logo/i).length).toBe(1);
  });
});
