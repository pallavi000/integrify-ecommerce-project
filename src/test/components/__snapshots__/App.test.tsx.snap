// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Should layout match snapshot 1`] = `
<body>
  <div>
    <div>
      <button>
        Create new user
      </button>
      <button>
        Empty users list
      </button>
      <button>
        Update one user
      </button>
      <button>
        Sort by email
      </button>
      <input
        name="search"
        type="text"
        value=""
      />
    </div>
  </div>
</body>
`;
